admin/src/main/java/edu/unc/lib/dl/admin/controller/PerformanceMonitorController.java

```diff
@@ -0,0 +1,299 @@
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+
+@Controller
+public class PerformanceMonitorController {
+	private static final Logger log = LoggerFactory
+			.getLogger(PerformanceMonitorController.class);
+	
+	private static final String NEW_LINE_SEPARATOR = "\n";
+	private static final Object [] FILE_HEADERS = {
+		"date",
+		"uuid",
+		"throughput_files",
+		"throughput_bytes",
+		"queued_duration",
+		"ingest_duration",
+		"finished",
+		"moves",
+		"finished_enhancements",
+		"failed_enhancements",
+		"failed_deposit",
+		"failed_deposit_job"
+	};
+	private CSVFormat csvFileFormat = CSVFormat.DEFAULT.withRecordSeparator(NEW_LINE_SEPARATOR);
+	
+	@Autowired
+	private JedisPool jedisPool;
+	
+	public JedisPool getJedisPool() {
+		return jedisPool;
+	}
+
+	public void setJedisPool(JedisPool jedisPool) {
+		this.jedisPool = jedisPool;
+	}
+
+	public Set<String> getDepositMetrics() {
+		try (Jedis jedis = getJedisPool().getResource()){
+			return jedis.keys("deposit-metrics:*");
+		}
+	}
+	
+	private Boolean buildFile(String path) {
+		File csvFile = new File(path);
+		
+		if(csvFile.exists() && !csvFile.isDirectory()) { 
+			Long currentTime = System.currentTimeMillis();
+			Long fileCreationTime = csvFile.lastModified();
+			
+			if ((currentTime - fileCreationTime ) > 3600000) {
+				return true;
+			}
+			return false;
+		} else if (!csvFile.exists()) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+	
+	/**
+	 * Gets totals for metrics that are only aggregated daily such as moves & enhancements
+	 * It also grabs throughput totals for older data, before these fields were set to measure performance at the individual deposit level
+	 * @return
+	 */
+	public String getOperationsData() {
+		FileWriter fileWriter = null;
+		CSVPrinter csvFilePrinter = null;
+		Set<String> operations = null;
+		String filePath = "/opt/data/ingest-times-daily.csv";
+		Map<String, String> depositJob = null;
+		Map<String, String> operationJob = null;
+		String[] depositKeys = null;
+
+		if (buildFile(filePath)) {
+			Set<String> deposits = getDepositMetrics();
+			try (Jedis jedis = getJedisPool().getResource()) {
+				operations = jedis.keys("operation-metrics:*");
+
+				fileWriter = new FileWriter(filePath);
```

Comment:
Generally its good to use the try() to create and autoclose resources when you can.  In this case, you could put all your writers and the jedis into the try together (and maybe end up with a new star wars movie), and not worry about closing them later, like this:
```java
try (
	Jedis jedis = getJedisPool().getResource();
	FileWriter fileWriter = new FileWriter(filePath);
	CSVPrinter csvFilePrinter = new CSVPrinter(fileWriter, csvFileFormat);
) { ```